### PR TITLE
Proof of Validator

### PR DESCRIPTION
**Pull request by thomaslavaur**
First implementation of PoV with Merkle trees in Circom.

### CODE REVIEW

**Comment by thomaslavaur** [comment]
Don't merge: need to add a verification part that the note used wasn't already spend

**Review by megonen — APPROVED** [review]
lgtm

**Comment by arnetheduck** [comment]
Is this described in a spec somewhere?

**Comment by thomaslavaur** [comment]
In the Proof of Validator [specification](<https://www.notion.so/Proof-of-Validator-Specification-5cd02f594d4d4a66be5e9f1ebf95d7d8>)

File: proof_of_validator/circom/validator_anemoi.circom [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+    root === hash[31].out * (1 - is_null);
+
+}
+
+template anemoi_proof_of_validator(max_notes, minimum_stake){   //TODO: put minimum_stake in the input to change it dynamically
+    signal input commitments_root;
+
+        // Note variables
+    signal input constraints[max_notes];    // Every note field represented as F_p elements for now (constraints are represented by their Merkle root)
```
[the remark this answers is not in the document]
Comment (thomaslavaur) [reply proof_of_validator/circom/validator_anemoi.circom:118]: It's constraint outside the circuit during the set construction

**Pull request merged**